server/pulp/server/managers/consumer/agent.py

```diff
@@ -404,6 +418,46 @@ def _unbindings(bindings):
             agent_bindings.append(agent_binding)
         return agent_bindings
 
+    @staticmethod
+    def delete_queue(url, name, consumer_id):
+        """
+        Delete the agent queue.
+        :param url: The broker URL.
+        :type url: str
+        :param name: The queue name.
+        :type name: str
+        :param consumer_id: The consumer ID.
+        :type consumer_id: str
+        """
+        try:
+            manager = managers.consumer_manager()
+            manager.get_consumer(consumer_id)
+            return  # still registered (abort)
+        except MissingResource:
+            # expected
+            pass
+        agent = PulpAgent()
+        agent.delete_queue(url, name)
+
+
+@task(base=Task)
+def delete_queue(url, name, consumer_id):
+    """
+    Task to delete the agent queue.
+    :param url: The broker URL.
+    :type url: str
+    :param name: The queue name.
+    :type name: str
+    :param consumer_id: The consumer ID.
+    :type consumer_id: str
+    """
+    try:
+        AgentManager.delete_queue(url, name, consumer_id)
+        logger.info(QUEUE_DELETED, {'name': name})
```

Comment:
This probably isn't a line you are guarding for exceptions. This should be moved to an 'else' clause so this would have try/except/else as codepath branches.